app/controllers/support_interface/bulk_upload/permissions_controller.rb

```diff
@@ -0,0 +1,81 @@
+module SupportInterface
+  module BulkUpload
+    class PermissionsController < SupportInterfaceController
+      def edit
+        multiple_provider_users_wizard = MultipleProviderUsersWizard.build(
+          state_store: multiple_provider_user_store,
+          provider_id: provider_id_param,
+          position: position_param,
+        )
+
+        @provider = Provider.find(provider_id_param)
+        @form = multiple_provider_users_wizard.single_provider_user_form(state_store_id: params[:id])
+        @position_and_count = multiple_provider_users_wizard.position_and_count(@form.state_store_id)
+        @backlink_path = backlink_path(@position_and_count, multiple_provider_users_wizard)
+      end
+
+      def update
+        multiple_provider_users_wizard = MultipleProviderUsersWizard.new(
+          state_store: multiple_provider_user_store,
+          provider_id: provider_id_param,
+        )
+
+        @provider = Provider.find(provider_id_param)
+        @form = CreateSingleProviderUserForm.new(
+          provider_user_params.merge(provider_permissions: provider_permissions_params),
+        )
+        @position_and_count = multiple_provider_users_wizard.position_and_count(@form.state_store_id)
+
+        if @form.valid?
+          multiple_provider_users_wizard.save_user_to_state_store!(@form)
+
+          if multiple_provider_users_wizard.more_users_to_process?
+            position = @position_and_count[:position] + 1
+            state_store_id = multiple_provider_users_wizard.state_store_id_for_position(position)
```

Comment:
I think we should avoid exposing the internals and state store id as a url param. Maybe we can update the state store id using the wizard, by invoking a method as so:
```
multiple_provider_users_wizard.increment_position!
```
which itself bumps the position and saves the state?

Reply:
Yes originally I did not expose the state store id as a url param but then this causes an issue for backlinks and looping through the Permissions forms.